### ISSUE TITLE
Target directory different case than .cf file name (departs from 2.7.0)

### ISSUE THREAD

**Issue by andreasabel** [BNFC/bnfc#120]
Bnfc 2.7.0 generated the files for CC.cf in a subdirectory named CC.  Bnfc 2.7.1 (development version) puts the generated files into directory Cc, modifying the case.  This is bad for case-sensitive file systems.

Maybe this change was not intended after all?  It would be good if the name of the directory would be literally the filename (without extension).

**Comment by gdetrez** [comment]
> Maybe this change was not intended after all? It would be good if the name of the directory would be literally the filename (without extension).

The change was intentional. The problem with using the literal filename is that not all valid filenames are valid haskell module names. For instance, ghc won't accept foobar or Foo-Bar, so this change was trying to transform them into the valid FooBar instead.

In your example I was trying to follow the convention that only the first letter in an abbreviation should be capitalized (so HttpModule and not HTTPModule). But that might be a wee bit too opinionated. ;-)

**Comment by andreasabel** [comment]
But CC is both a valid filename and a haskell module name, so why has it to be touched at all?

It becomes very difficult to write a Makefile that works immediately, since I cannot predict what BNFC will do the the input file name.  Also, depending on the installed BNFC version, the Makefile has to change.

Here is a sample Makefile, observe how I need a variable `what_bnfc_makes_out_of_grm` that simulates BNFC's name touching.
```make
grm = CPP
what_bnfc_makes_out_of_grm = Cpp
dir = bnfc/$(what_bnfc_makes_out_of_grm)
bnfc_output = $(patsubst %,$(dir)/%,Abs.hs ErrM.hs Print.hs Lex.x Par.y Test.hs)
alex_file = $(dir)/Lex
happy_file = $(dir)/Par
test_file = $(dir)/Test

build : $(alex_file).hs $(happy_file).hs

test : $(test_file)
	$(test_file) test/hello.cc
	$(test_file) test/greet.cc
	$(test_file) test/med.cc
	$(test_file) test/grade.cc
	$(test_file) test/palin.cc
	$(test_file) test/grammar.cc

$(bnfc_output): src/$(grm).cf
	-@mkdir -p bnfc
	-@rm $(bnfc_output)
	@(cd bnfc && bnfc -d ../$<)

$(test_file) : $(bnfc_output) $(alex_file).hs $(happy_file).hs
	@(cd bnfc && ghc $(what_bnfc_makes_out_of_grm)/Test.hs)

$(alex_file).hs: $(alex_file).x
	alex $<

$(happy_file).hs: $(happy_file).y
	happy -iGRM.txt $<
```

**Comment by gdetrez** [comment]
> But CC is both a valid filename and a haskell module name, so why has it to be touched at all?

It shouldn't be touched and it should be fixed by commit e37b7f5.

**Comment by andreasabel** [comment]
My apologies, something must have gone wrong with cabal install after pulling such that I was still using an old bnfc.  I can confirm that the problem is fixed! 
Thanks!
